### PR TITLE
Add: support a fully parseable JSON output.

### PR DESCRIPTION
**Pull request by juanefec**
This PR adds a way to have a parseable json output.

 - New prop on Options to separate changed values.
 - New DefaultJSONOptions() for an easier access to the feature.

### CODE REVIEW

File: jsondiff.go [hunk 1]
```diff
@@ -43,34 +43,48 @@ type Tag struct {
...
-	Prefix     string
-	Indent     string
-	PrintTypes bool
+	Normal           Tag
+	Added            Tag
+	Removed          Tag
+	Changed          Tag
+	Prefix           string
+	Indent           string
+	PrintTypes       bool
```
Comment (fabzter) [inline jsondiff.go:52]: maybe new indentation is unnecessary, let's wait check what @nsf says

Comment (nsf) [reply jsondiff.go:52]: Well, it's a result of gofmt, it's fine. Also the whole pull request looks fine too, will merge it in a sec.

**Comment by nsf** [comment]
While I merged it, but it looks as innocent hack at best. I understand the need for structured output, however why not do it in Go for library consumers in Go language. That's probably a proper way to do it. But I understand people can be lazy (as I am myself). And if hack works for a particular use case, sure.

**Comment by nsf** [comment]
In any case. Thanks for the patch!

**Pull request merged**